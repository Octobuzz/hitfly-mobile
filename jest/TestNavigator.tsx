// https://github.com/react-navigation/rfcs/issues/75
import React, { Children } from 'react'
import {
  createAppContainer,
  createSwitchNavigator,
  NavigationParams,
} from 'react-navigation'

const TestNavigator = ({
  children,
  params,
}: {
  children: any
  params?: NavigationParams
}) => {
  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: { screen: () => Children.only(children), params },
    }),
  )
  return <Navigator />
}

export default TestNavigator
