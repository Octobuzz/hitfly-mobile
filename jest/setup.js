import 'jest-styled-components'
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock'

jest.mock('react-native-vector-icons/Foundation', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')
jest.mock('react-native-vector-icons/Feather', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')
jest.mock('react-native-vector-icons/SimpleLineIcons', () => 'Icon')

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)

jest.mock('react-native-track-player', () => {
  return {
    addEventListener: jest.fn(),
    registerEventHandler: jest.fn(),
    registerPlaybackService: jest.fn(),
    setupPlayer: jest.fn(),
    destroy: jest.fn(),
    updateOptions: jest.fn(),
    add: jest.fn(),
    remove: jest.fn(),
    skip: jest.fn(),
    skipToNext: jest.fn(),
    skipToPrevious: jest.fn(),
    removeUpcomingTracks: jest.fn(),
    // playback commands
    reset: jest.fn(),
    play: jest.fn(),
    pause: jest.fn(),
    stop: jest.fn(),
    seekTo: jest.fn(),
    setVolume: jest.fn(),
    setRate: jest.fn(),
    // player getters
    getQueue: jest.fn(),
    getTrack: jest.fn(),
    getCurrentTrack: jest.fn(),
    getVolume: jest.fn(),
    getDuration: jest.fn(),
    getPosition: jest.fn(),
    getBufferedPosition: jest.fn(),
    getState: jest.fn(),
    getRate: jest.fn(),
  }
})

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})
