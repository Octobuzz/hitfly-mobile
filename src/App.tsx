import React from 'react'
import './reactotron.config'
import { YellowBox } from 'react-native'
import { ThemeProvider } from 'styled-components'
import ApolloClient, { InMemoryCache } from 'apollo-boost'
import { persistCache } from 'apollo-cache-persist'
import { ApolloProvider } from '@apollo/react-hooks'
import AppNavigator from './navigation'
import { storageInstance, storage } from './utils'
import theme from './theme'
import { names, storageKeys } from './constants'

// либы до сих пор используют это
YellowBox.ignoreWarnings([
  'Warning: componentWillMount is deprecated',
  'Warning: componentWillUpdate is deprecated',
  'Warning: componentWillReceiveProps is deprecated',
])

async function createApolloClient(): Promise<ApolloClient<InMemoryCache>> {
  const cache = new InMemoryCache({})

  await persistCache({
    cache,
    // @ts-ignore
    storage: storageInstance,
  })

  const token = await storage.getItem(storageKeys.AUTH_TOKEN)

  const client = new ApolloClient<InMemoryCache>({
    cache,
    headers: {
      'X-TOKEN-AUTH': token,
    },
    uri: names.BASE_URL,
  })

  return client
}

class App extends React.Component<{}, { isReady: boolean }> {
  client?: ApolloClient<InMemoryCache>
  state = {
    isReady: false,
  }

  async componentDidMount() {
    this.client = await createApolloClient()
    this.setState({ isReady: true })
  }

  render() {
    const { isReady } = this.state
    return (
      isReady &&
      this.client && (
        <ApolloProvider client={this.client}>
          <ThemeProvider theme={theme}>
            <AppNavigator />
          </ThemeProvider>
        </ApolloProvider>
      )
    )
  }
}

export default App
