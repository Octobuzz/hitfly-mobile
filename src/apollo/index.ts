import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
