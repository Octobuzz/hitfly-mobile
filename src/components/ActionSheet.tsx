import React, { forwardRef, Ref } from 'react'
import { StyleSheet } from 'react-native'
import {
  ActionSheetCustom,
  ActionSheetCustomProps,
} from 'react-native-actionsheet'
import { getBottomSpace } from 'react-native-iphone-x-helper'
import theme from 'src/theme'

interface Props extends ActionSheetCustomProps {
  // TS говно, не захотел подхватить ref
  ref: Ref<ActionSheetCustom>
}

export type ActionSheetInstance = ActionSheetCustom

const ActionSheet: React.FC<Props> = forwardRef<ActionSheetCustom, Props>(
  (props, ref) => {
    return (
      <ActionSheetCustom
        {...props}
        buttonUnderlayColor={theme.colors.buttonUnderlayColor}
        tintColor={theme.colors.white}
        ref={ref}
        styles={styles}
      />
    )
  },
)

// Для библиотеки так
const styles = StyleSheet.create({
  wrapper: { paddingBottom: getBottomSpace() },
  body: {
    backgroundColor: theme.colors.black,
  },
  messageText: {
    color: theme.colors.white,
    fontSize: 14,
    textAlign: 'center',
  },
  messageBox: {
    height: 50,
    paddingBottom: 0,
    backgroundColor: theme.colors.black,
  },
  buttonBox: {
    height: 50,
    backgroundColor: theme.colors.black,
  },
  buttonText: {
    color: theme.colors.white,
  },
  cancelButtonBox: {
    height: 50,
    marginTop: 6,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.black,
  },
})

export default ActionSheet
