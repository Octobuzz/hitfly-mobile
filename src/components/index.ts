export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
