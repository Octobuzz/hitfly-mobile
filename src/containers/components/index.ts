export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
