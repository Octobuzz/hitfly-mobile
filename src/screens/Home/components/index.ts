export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
