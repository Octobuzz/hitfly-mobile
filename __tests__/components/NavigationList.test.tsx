import React from 'react'
import { render, fireEvent } from '../../jest/test-utils'
import { NavigationList } from 'src/components'

describe('NavigationList', () => {
  it('renders correctly with props', () => {
    const props = {
      items: [
        {
          title: 'title',
          onPress: jest.fn(),
        },
      ],
    }

    const { getAllByTestId } = render(<NavigationList {...props} />)

    fireEvent.press(getAllByTestId('item')[0])
    expect(props.items[0].onPress).toBeCalled()
  })

  it('renders correctly with empty items', () => {
    const { asJSON } = render(<NavigationList items={[]} />)

    expect(asJSON()).toMatchSnapshot()
  })

  it('renders correctly with 2 items', () => {
    const props = {
      items: [
        {
          title: 'title',
          onPress: jest.fn(),
        },
        {
          title: 'title2',
          onPress: jest.fn(),
        },
      ],
    }
    const { asJSON } = render(<NavigationList {...props} />)

    expect(asJSON()).toMatchSnapshot()
  })
})
