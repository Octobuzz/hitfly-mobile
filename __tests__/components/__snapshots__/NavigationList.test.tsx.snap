// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationList renders correctly with 2 items 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#E4E4E4",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingVertical": 28,
          },
        ]
      }
      testID="item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "GothamPro",
              "fontSize": 16,
              "lineHeight": 20,
              "paddingRight": 16,
            },
          ]
        }
      >
        title
      </Text>
      <Icon
        name="arrow-forward"
        size={20}
        style={
          Array [
            Object {
              "color": "#313131",
            },
          ]
        }
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#E4E4E4",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingVertical": 28,
          },
        ]
      }
      testID="item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "GothamPro",
              "fontSize": 16,
              "lineHeight": 20,
              "paddingRight": 16,
            },
          ]
        }
      >
        title2
      </Text>
      <Icon
        name="arrow-forward"
        size={20}
        style={
          Array [
            Object {
              "color": "#313131",
            },
          ]
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`NavigationList renders correctly with empty items 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  />
</View>
`;
